gen/DocumentFormat.OpenXml.Generator.Models/Generators/Elements/DataModelWriterExtensions.cs

```diff
@@ -100,7 +240,16 @@ private static void WriteType(this IndentedTextWriter writer, OpenXmlGeneratorSe
             foreach (var attribute in element.Attributes)
             {
                 delimiter.AddDelimiter();
-                writer.WriteAttributeProperty(services, attribute);
+
+                if (_attributeData.TryGetValue(element.Name.ToString(), out Dictionary<string, List<string>> attrAttributeData)
```

Comment:
you could make the dictionary have the type of element.Name (I think it's a `TypedQName`) and it'll make it a bit more performant. The keys you're creating will also automatically be converted to it as well

Reply:
thanks, that was a good idea. for `SchemaAttributes`, they aren't actually `TypedQName` but I decided to try using the `Type` property of the attribute metadata to construct a string that can convert to a `TypedQName` for the purposes of keys in the dictionary. It was either that or `string.Empty` which would have issues in parsing. So even though the `Type` property of the attribute metadata is actually the c# type (class), it works. If you have an alternative suggestion I'll try that.